pkg/controllers/queue/queue_controller.go

```diff
@@ -279,12 +280,21 @@ func (c *Controller) deletePodGroup(obj interface{}) {
 		}
 	}
 
-	key, _ := cache.MetaNamespaceKeyFunc(obj)
+	key, _ := cache.MetaNamespaceKeyFunc(pg)
+	queue := getPogGroupQueue(pg)
 
 	c.pgMutex.Lock()
 	defer c.pgMutex.Unlock()
 
-	delete(c.podGroups[pg.Spec.Queue], key)
+	delete(c.podGroups[queue], key)
 
-	c.queue.Add(pg.Spec.Queue)
+	c.queue.Add(queue)
+}
+
+func getPogGroupQueue(pg *schedulingv1alpha2.PodGroup) string {
+	if pg.Spec.Queue == "" {
```

Comment:
prefer to handle this in admission controller.

Reply:
We can mutate in admission webhook, but keep in mind admission is an optional component.

We do not have admission for pg now, it is a big cost just for a default value set. Anyway if you really insist on that,   crd definition can set default value.